datalake/src/main/java/com/sequenceiq/datalake/service/sdx/attach/SdxDetachNameGenerator.java

```diff
@@ -0,0 +1,33 @@
+package com.sequenceiq.datalake.service.sdx.attach;
+
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.stereotype.Component;
+
+/**
+ * Provides a central logic point for generation of a detached SDX cluster name, as
+ * well as reversing a detached name.
+ */
+@Component
+public class SdxDetachNameGenerator {
+    static final Pattern DETACHED_NAME_PATTERN = Pattern.compile(".*(-[0-9]+)$");
+
+    private static final String DELIMITER = "-";
+
+    String generateDetachedClusterName(String originalName) {
```

Comment:
public please and for the next method too

Reply:
Sure I can do this.

Reply:
Done.